app/util/planParamUtil.js

```diff
@@ -255,6 +255,17 @@ export const preparePlanParams = (config, useDefaultModes) => (
     intermediatePlaceLocations,
   );
 
+  // Use defaults or user given settings
+  const ticketTypes = useDefaultModes
```

Comment:
This change effectively changes the meaning of the `useDefaultModes` variable to "use default modes and some other selected (default) routing parameters". This subtle change confused us quite a bit.

---

The DX aside, what is the intended UX? The `useDefaultModes = true` query may return itineraries which the user did not expect?

> We need to treat ticket types as modes when querying with default settings, i.e. don't include ticket type in all modes query.

Why is that? (I don't understand the reasoning here, probably because I don't have some context knowledge.)

Maybe `useRelaxedRoutingPreferences` would be a more appropriate naming?

cc @vesameskanen

Reply:
I updated the name in v3.